Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -2309,88 +2309,132 @@ ThrowCompletionOr<TriState> is_less_than(VM& vm, Value lhs, Value rhs, bool left
     Value x_primitive;
     Value y_primitive;
 
+    // 1. If the LeftFirst flag is true, then
     if (left_first) {
+        // a. Let px be ? ToPrimitive(x, number).
         x_primitive = TRY(lhs.to_primitive(vm, Value::PreferredType::Number));
+
+        // b. Let py be ? ToPrimitive(y, number).
         y_primitive = TRY(rhs.to_primitive(vm, Value::PreferredType::Number));
     } else {
+        // a. NOTE: The order of evaluation needs to be reversed to preserve left to right evaluation.
+
+        // b. Let py be ? ToPrimitive(y, number).
         y_primitive = TRY(lhs.to_primitive(vm, Value::PreferredType::Number));
+
+        // c. Let px be ? ToPrimitive(x, number).
         x_primitive = TRY(rhs.to_primitive(vm, Value::PreferredType::Number));
     }
 
+    // 3. If px is a String and py is a String, then
     if (x_primitive.is_string() && y_primitive.is_string()) {
         auto x_string = x_primitive.as_string().deprecated_string();
         auto y_string = y_primitive.as_string().deprecated_string();
 
         Utf8View x_code_points { x_string };
         Utf8View y_code_points { y_string };
 
-        if (x_code_points.starts_with(y_code_points))
-            return TriState::False;
-        if (y_code_points.starts_with(x_code_points))
-            return TriState::True;
-
```

Comment:
Already rebased on the other PR? There could be conflicts in your near future

Reply:
Fixed now :+1: